### PR TITLE
Document ICanvas

### PR DESCRIPTION
**Pull request by jknaudt21**
### Description of Change

Added the missing docs for the `ICanvas` interface

Docs were based these conceptual docs to maintain alignment: 
- https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/draw?view=net-maui-8.0
- https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/transforms?view=net-maui-8.0
- https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/?view=net-maui-8.0
- https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/paint?view=net-maui-8.0

---

Part of documentation effort, related to https://github.com/dotnet/maui/issues/3960

### CODE REVIEW

File: src/Graphics/src/Graphics/ICanvas.cs [hunk 1]
```diff
@@ -62,37 +244,117 @@ public interface ICanvas
...
 		public void DrawImage(IImage image, float x, float y, float width, float height);
 
+		/// <summary>
+		/// Calculates the are a string would occupy if drawn on the canvas.
+		/// </summary>
+		/// <param name="value">Text to calculate the size on.</param>
+		/// <param name="font"></param>
+		/// <param name="fontSize"></param>
+		/// <returns>The size of </returns>
 		public SizeF GetStringSize(string value, IFont font, float fontSize);
```
Comment (jknaudt21) [inline src/Graphics/src/Graphics/ICanvas.cs:347]: @mattleibow please confirm that I interpreted this method correctly.

File: src/Graphics/src/Graphics/ICanvas.cs [hunk 2]
```diff
@@ -62,37 +244,117 @@ public interface ICanvas
...
+		/// <param name="height">Height of the image.</param>
 		public void DrawImage(IImage image, float x, float y, float width, float height);
 
+		/// <summary>
+		/// Calculates the area a string would occupy if drawn on the canvas.
+		/// </summary>
+		/// <param name="value">String to calculate the size on.</param>
+		/// <param name="font">The string's font type.</param>
+		/// <param name="fontSize">The string's font size.</param>
+		/// <returns>The size of </returns>
```
Comment (jfversluis) [inline src/Graphics/src/Graphics/ICanvas.cs:346]: Returns the size of......?

File: src/Graphics/src/Graphics/ICanvas.cs [hunk 3]
```diff
@@ -62,37 +244,117 @@ public interface ICanvas
...
 
+		/// <summary>
+		/// Calculates the area a string would occupy if drawn on the canvas.
+		/// </summary>
+		/// <param name="value">String to calculate the size on.</param>
+		/// <param name="font">The string's font type.</param>
+		/// <param name="fontSize">The string's font size.</param>
+		/// <param name="horizontalAlignment">Horizontal alignment options for the string.</param>
+		/// <param name="verticalAlignment">Vertical alignment options for the string.</param>
+		/// <returns></returns>
```
Comment (jfversluis) [inline src/Graphics/src/Graphics/ICanvas.cs:357]: What does this return?

**Review by jfversluis — CHANGES_REQUESTED** [review]

**Review by jsuarezruiz — APPROVED** [review]

File: src/Graphics/src/Graphics/ICanvas.cs [hunk 4]
```diff
@@ -3,54 +3,236 @@
 
 namespace Microsoft.Maui.Graphics
 {
+	/// <summary>
+	/// Represents a platform-agnostic canvas on which 2D graphics can be drawn using types from the <see cref="Microsoft.Maui.Graphics"/> namespace.
+	/// </summary>
 	public interface ICanvas
 	{
+		/// <summary>
+		/// Gets or sets a value that represents the scaling factor to scale the UI by.
```
Comment (mattleibow) [inline src/Graphics/src/Graphics/ICanvas.cs:12]: Scale the UI? Maybe drawing or canvas operations?

File: src/Graphics/src/Graphics/ICanvas.cs (hunk 1 above)
Comment (mattleibow) [reply src/Graphics/src/Graphics/ICanvas.cs:347]: Correct. This "measures" the string.

**Review by mattleibow — APPROVED** [review]

File: src/Graphics/src/Graphics/ICanvas.cs (hunk 4 above)
Comment (jknaudt21) [reply src/Graphics/src/Graphics/ICanvas.cs:12]: This comes from a ported conceptual doc: https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/?view=net-maui-8.0

I like the sound of "... scale the canvas operations"

**Review by jfversluis — APPROVED** [review]

**Pull request merged**